controllers/handlers/passt/passt.go

```diff
@@ -0,0 +1,345 @@
+package passt
+
+import (
+	"fmt"
+	"maps"
+	"os"
+	"sync"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	securityv1 "github.com/openshift/api/security/v1"
+
+	kubevirtcorev1 "kubevirt.io/api/core/v1"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/operands"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/nodeinfo"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	DeployPasstNetworkBindingAnnotation = "deployPasstNetworkBinding"
+
+	BindingName = "passt"
+
+	networkBindingNADName       = "primary-udn-kubevirt-binding"
+	networkBindingNADNamespace  = "default"
+	NetworkAttachmentDefinition = networkBindingNADNamespace + "/" + networkBindingNADName
+
+	bindingComputeMemoryOverhead = "250Mi"
+)
+
+var (
+	passtResourceMemory = resource.MustParse(bindingComputeMemoryOverhead)
+	passtImage          string
+	passtImageOnce      sync.Once
+)
+
+// CheckPasstImagesEnvExists checks if the passt image environment variable exists
+func CheckPasstImagesEnvExists() error {
+	if _, passtImageVarExists := os.LookupEnv(hcoutil.PasstImageEnvV); !passtImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstImageEnvV)
+	}
+	if _, passtCNIImageVarExists := os.LookupEnv(hcoutil.PasstCNIImageEnvV); !passtCNIImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstCNIImageEnvV)
+	}
+	return nil
+}
+
+// NetworkBinding creates an InterfaceBindingPlugin for passt network binding
+func NetworkBinding(sidecarImage string) kubevirtcorev1.InterfaceBindingPlugin {
+	return kubevirtcorev1.InterfaceBindingPlugin{
+		NetworkAttachmentDefinition: NetworkAttachmentDefinition,
+		SidecarImage:                sidecarImage,
+		Migration:                   &kubevirtcorev1.InterfaceBindingMigration{},
+		ComputeResourceOverhead: &kubevirtcorev1.ResourceRequirementsWithoutClaims{
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: passtResourceMemory,
+			},
+		},
+	}
+}
+
+// GetImage gets the passt image from environment variable
+func GetImage() string {
+	passtImageOnce.Do(func() {
+		passtImage = os.Getenv(hcoutil.PasstImageEnvV)
+	})
+	return passtImage
+}
```

Comment:
not sure we need this function at all. we can just read the env var when defining the passtImage variable, in line 43. Then we won't need the syncOnce either.